sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/ColumnValue.java

```diff
@@ -193,7 +193,7 @@ public static TColumnValue toTColumnValue(Type type, Object value) {
     case INTERVAL_DAY_TIME_TYPE:
       return stringValue((HiveIntervalDayTime) value);
     case DECIMAL_TYPE:
-      return stringValue(((HiveDecimal)value));
```

Comment:
Sorry for late response, @sujith71955 . But, this PR is difficult to validate.

BTW, Hive seems to choose a different path; [here](https://github.com/apache/hive/blob/master/service/src/java/org/apache/hive/service/cli/ColumnValue.java#L210-L211) and [here](https://github.com/apache/hive/blob/master/service/src/java/org/apache/hive/service/cli/ColumnValue.java#L151-L158)
```java
    case DECIMAL_TYPE:
      return stringValue((HiveDecimal)value, typeDescriptor);
```

```java
  private static TColumnValue stringValue(HiveDecimal value, TypeDescriptor typeDescriptor) {
    TStringValue tStrValue = new TStringValue();
    if (value != null) {
      int scale = typeDescriptor.getDecimalDigits();
      tStrValue.setValue(value.toFormatString(scale));
    }
    return TColumnValue.stringVal(tStrValue);
  }
```

Since this PR reports a bug with `Hive ODBC Client`, I'm wondering if Hive works correctly with Hive ODBC Driver with the above code? If then, isn't it more natural to follow the upstream?

Reply:
i will explore more based on your comment and update you guys.Thanks for your valuable suggestion.